src/main/groovy/edu/oregonstate/mist/personsapi/mapper/PersonMapper.groovy

```diff
@@ -14,6 +14,17 @@ import java.sql.SQLException
 public class PersonMapper implements ResultSetMapper<PersonObject> {
     PhoneFormatter phoneFormatter = new PhoneFormatter()
     public PersonObject map(int i, ResultSet rs, StatementContext sc) throws SQLException {
+        String rawSSN = rs.getString('SSN')
+        String ssnStatus = 'invalid'
+
+        if (rawSSN && rawSSN.startsWith('VAULT')) {
+            ssnStatus = 'valut'
+        } else if (rawSSN == 'REDACTED' || rawSSN == 'ARCHIVED') {
+            ssnStatus = rawSSN.toLowerCase()
+        } else if (rawSSN && rawSSN.matches(/^(?!0{9})\d{9}$/)) {
+            ssnStatus = 'valid'
+        }
```

Comment:
Would it make sense to move this logic to a private method of this class, similar to `formatPhoneNumber()`?

Reply:
yeah, you're right, will do.